core/src/avm2/globals/flash/display/BitmapData.as

```diff
@@ -68,6 +69,10 @@ package flash.display {
         ):int;
 
         public function generateFilterRect(sourceRect:Rectangle, filter:BitmapFilter):Rectangle {
+            // Flash always reports that a ShaderFilter affects the entire BitampData, ignoring SourceRect.
```

Comment:
Can you add a test for this in BitmapData? Also, does this ignore the ShaderFilter extensions?

Reply:
There's already a test for this in `pixelbender_effect_glassDisplace_shaderfilter`